### PR TITLE
Take reds tax into account for oxygen. Clean things up.

### PR DESCRIPTION
**Pull request by kberg**

### CODE REVIEW

File: src/Player.ts [hunk 1]
```diff
@@ -1861,25 +1861,33 @@ export class Player implements ISerializable<SerializedPlayer> {
...
+    tr = {...tr};
     let total = tr.tr ?? 0;
 
+    if (tr.oxygen !== undefined) {
+      const availableSteps = constants.MAX_OXYGEN_LEVEL - this.game.getOxygenLevel();
+      const steps = Math.min(availableSteps, tr.oxygen);
+      total = total + steps;
+      // TODO(kberg): Add constants for these constraints.
+      if (this.game.getOxygenLevel() < 8 && this.game.getOxygenLevel() + steps >= 8) {
+        tr.temperature = (tr.temperature ?? 0) + 1;
```
Comment (kberg) [inline src/Player.ts:1877]: By making a local copy and incrementing values, it takes into account whether those global parameters actually go up that high.

File: tests/cards/base/StripMine.spec.ts [hunk 2]
```diff
@@ -45,16 +45,15 @@ describe('StripMine', function() {
...
-    const reds = new Reds();
-    turmoil.rulingParty = reds;
+    turmoil.rulingParty = new Reds();
     PoliticalAgendas.setNextAgenda(turmoil, game);
-    expect(card.canPlay(player)).is.false;
+    expect(player.canPlay(card)).is.false;
 
     player.megaCredits += REDS_RULING_POLICY_COST * 2; // Payment for Reds tax
-    expect(card.canPlay(player)).is.true;
+    expect(player.canPlay(card)).is.true;
```
Comment (kberg) [inline tests/cards/base/StripMine.spec.ts:53]: YES!

File: src/Player.ts [hunk 3]
```diff
@@ -1858,28 +1861,38 @@ export class Player implements ISerializable<SerializedPlayer> {
       (canUseScience ? this.getSpendableScienceResources() : 0);
   }
 
+  // TODO(kberg): Move this to somewhere in turmoil/
```
Comment (bafolts) [inline src/Player.ts:1864]: Yes please.

**Review by bafolts — APPROVED** [review]

File: src/Player.ts [hunk 4]
```diff
@@ -1848,8 +1848,11 @@ export class Player implements ISerializable<SerializedPlayer> {
     const canUseMicrobes: boolean = options?.microbes ?? false;
     const canUseScience: boolean = options?.science ?? false;
 
-    return cost <=
-    this.megaCredits - (reserveUnits.megacredits + (options?.redsCost ?? 0)) +
+    const availableMegacredits = this.megaCredits - (reserveUnits.megacredits + (options?.redsCost ?? 0));
+    if (availableMegacredits < 0) {
+      return false;
+    }
+    return cost <= availableMegacredits +
```
[the remark this answers is not in the document]
Comment (kberg) [reply src/Player.ts:1855]: Are you forgetting the `+` at the end of the line?

**Pull request merged**